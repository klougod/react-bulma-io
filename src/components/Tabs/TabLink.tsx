import React, { ReactNode, MouseEvent } from 'react'

interface TabLinkProps {
  className?: string
  id?: string
  text?: ReactNode
  isActive?: boolean
  changeTab?: (e: MouseEvent<HTMLAnchorElement>) => void
  children?: ReactNode
}

export const TabLink = ({ isActive, text, id, changeTab }: TabLinkProps) => {
  return (
    <li className={isActive ? 'is-active' : ''}>
      {/* eslint-disable-next-line */}
      <a
        id={id}
        className='is-size-4-tablet is-size-6-mobile'
        style={{ padding: '0.5em 0.2em !important' }}
        onClick={changeTab}
      >
        {text}
      </a>
    </li>
  )
}

TabLink.defaultProps = {
  className: '',
  id: '',
  text: '',
  isActive: false,
  changeTab: () => {}
}

export default TabLink
