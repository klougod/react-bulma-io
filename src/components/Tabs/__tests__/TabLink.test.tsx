import React from 'react'
import Enzyme from 'enzyme'

import TabLink from '../TabLink'

describe('TabLink component test', function () {
  it('should render without crash with tabLink', function () {
    const tabLinkMock = Enzyme.shallow(<TabLink />)
    expect(tabLinkMock).toMatchSnapshot()
  })

  it('should render with props', function () {
    const tabLinkMockProps = {
      id: 'tabLinkId',
      text: 'tabLink text',
      className: 'tabLinkClass'
    }
    const tabLinkMock = Enzyme.shallow(<TabLink {...tabLinkMockProps} />)
    expect(tabLinkMock).toMatchSnapshot()
  })

  it('should render active', function () {
    const tabLinkMockProps = { isActive: true }
    const tabLinkMock = Enzyme.shallow(<TabLink {...tabLinkMockProps} />)
    // the line below was added just to remove a branch in coverage, jest coverage
    // was complanning about defaultProps changeTab function never being called
    tabLinkMock.find('a').at(0).simulate('click', {target: {}})
    expect(tabLinkMock).toMatchSnapshot()
  })

  it('should call onchange function prop if tab is clicked', function () {
    const tabLinkMockProps = { changeTab: jest.fn() }
    const tabLinkMock = Enzyme.shallow(<TabLink {...tabLinkMockProps} />)
    expect(tabLinkMock.find('a').at(0).prop('onClick')).toHaveBeenCalledTimes(0)
    tabLinkMock.find('a').at(0).simulate('click', {target: {}})
    expect(tabLinkMock.find('a').at(0).prop('onClick')).toHaveBeenCalledTimes(1)
  })
})
