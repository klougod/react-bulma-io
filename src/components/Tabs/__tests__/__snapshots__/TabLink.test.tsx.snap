// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLink component test should render active 1`] = `
<li
  className="is-active"
>
  <a
    className="is-size-4-tablet is-size-6-mobile"
    id=""
    onClick={[Function]}
    style={
      Object {
        "padding": "0.5em 0.2em !important",
      }
    }
  />
</li>
`;

exports[`TabLink component test should render with props 1`] = `
<li
  className=""
>
  <a
    className="is-size-4-tablet is-size-6-mobile"
    id="tabLinkId"
    onClick={[Function]}
    style={
      Object {
        "padding": "0.5em 0.2em !important",
      }
    }
  >
    tabLink text
  </a>
</li>
`;

exports[`TabLink component test should render without crash with tabLink 1`] = `
<li
  className=""
>
  <a
    className="is-size-4-tablet is-size-6-mobile"
    id=""
    onClick={[Function]}
    style={
      Object {
        "padding": "0.5em 0.2em !important",
      }
    }
  />
</li>
`;
